lang/test/org/partiql/lang/planner/PlannerPipelineSmokeTests.kt

```diff
@@ -30,23 +28,46 @@ class PlannerPipelineSmokeTests {
     @Suppress("DEPRECATION")
     private fun createPlannerPipelineForTest(
         allowUndefinedVariables: Boolean,
+        plannerEventCallback: PlannerEventCallback?,
         block: PlannerPipeline.Builder.() -> Unit = { }
     ) = PlannerPipeline.build(ION) {
         allowUndefinedVariables(allowUndefinedVariables)
         globalVariableResolver(createFakeGlobalsResolver("Customer" to "fake_uid_for_Customer"))
+        plannerEventCallback?.let { plannerEventCallback(it) }
         block()
     }
 
     @Test
     fun `happy path`() {
-        val pipeline = createPlannerPipelineForTest(allowUndefinedVariables = true)
-        val result = pipeline.plan("SELECT c.* FROM Customer AS c WHERE c.primaryKey = 42")
+        var pecCallbacks = 0
+        val plannerEventCallback: PlannerEventCallback = { _ ->
+            pecCallbacks++
+            // println("*******************************************************************************")
+            // println(event)
```

Comment:
If you uncomment them you'll see exactly that the different `PlannerEventCallback`s are.

Reply:
I see. But still, is it a generally a good practice to remove commented code when merging changes to `main`?

Reply:
```
*******************************************************************************
event:    parse_sql
duration: PT0.068414S
input:
SELECT c.* FROM Customer AS c WHERE c.primaryKey = 42
output:
(query (select (project (project_list (project_all (id c (case_insensitive) (unqualified))))) (from (scan (id Customer (case_insensitive) (unqualified)) c null null)) (where (eq (path (id c (case_insensitive) (unqualified)) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)))))
*******************************************************************************
event:    normalize_ast
duration: PT0.02053S
input:
(query (select (project (project_list (project_all (id c (case_insensitive) (unqualified))))) (from (scan (id Customer (case_insensitive) (unqualified)) c null null)) (where (eq (path (id c (case_insensitive) (unqualified)) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)))))
output:
(query (select (project (project_list (project_all (id c (case_insensitive) (unqualified))))) (from (scan (id Customer (case_insensitive) (unqualified)) c null null)) (where (eq (path (id c (case_insensitive) (unqualified)) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)))))
*******************************************************************************
event:    ast_to_logical
duration: PT0.018401S
input:
(query (select (project (project_list (project_all (id c (case_insensitive) (unqualified))))) (from (scan (id Customer (case_insensitive) (unqualified)) c null null)) (where (eq (path (id c (case_insensitive) (unqualified)) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)))))
output:
(plan (stmt (query (bindings_to_values (struct (struct_fields (id c (case_insensitive) (unqualified)))) (filter (eq (path (id c (case_insensitive) (unqualified)) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)) (scan (id Customer (case_insensitive) (unqualified)) (var_decl c) null null))))) (version '0.0'))
*******************************************************************************
event:    logical_to_logical_resolved
duration: PT0.032425S
input:
(plan (stmt (query (bindings_to_values (struct (struct_fields (id c (case_insensitive) (unqualified)))) (filter (eq (path (id c (case_insensitive) (unqualified)) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)) (scan (id Customer (case_insensitive) (unqualified)) (var_decl c) null null))))) (version '0.0'))
output:
(plan (stmt (query (bindings_to_values (struct (struct_fields (local_id 0))) (filter (eq (path (local_id 0) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)) (scan (global_id fake_uid_for_Customer) (var_decl 0) null null))))) (version '0.0') (locals (local_variable c 0)))
*******************************************************************************
event:    logical_resolved_to_default_physical
duration: PT0.02509S
input:
(plan (stmt (query (bindings_to_values (struct (struct_fields (local_id 0))) (filter (eq (path (local_id 0) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)) (scan (global_id fake_uid_for_Customer) (var_decl 0) null null))))) (version '0.0') (locals (local_variable c 0)))
output:
(plan (stmt (query (bindings_to_values (struct (struct_fields (local_id 0))) (filter (impl default) (eq (path (local_id 0) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)) (scan (impl default) (global_id fake_uid_for_Customer) (var_decl 0) null null))))) (version '0.0') (locals (local_variable c 0)))
*******************************************************************************
event:    compile
duration: PT0.024945S
input:
(plan (stmt (query (bindings_to_values (struct (struct_fields (local_id 0))) (filter (impl default) (eq (path (local_id 0) (path_expr (lit "primaryKey") (case_insensitive))) (lit 42)) (scan (impl default) (global_id fake_uid_for_Customer) (var_decl 0) null null))))) (version '0.0') (locals (local_variable c 0)))
output:
Success(output=org.partiql.lang.planner.PlannerPipelineImpl$compile$1$queryPlan$2@21c3a271, warnings=[])

```

Reply:
But yes, generally it is a good idea.  However, I often find myself adding and removing this kind of commented line in code a lot so sometimes I just leave for the sake of convenience. 

I do *not* however ever intend to leave behind multiple lines of commented code just because "we might need it later".  

That said, this is small enough it doesn't save me much to leave it behind when I will inevitably need to add it back later during local development.  I'll remove it.

Reply:
I think probably you can add this test to the `examples/` folder?

Reply:
Oh, that's not a bad idea for #676, but for this little thing I don't think it's needed.